src/main/java/org/threadly/litesockets/Client.java

```diff
@@ -3,557 +3,358 @@
 
-import org.threadly.concurrent.SubmitterExecutorInterface;
-import org.threadly.litesockets.SocketExecuterBase.WireProtocol;
+import org.threadly.concurrent.future.ListenableFuture;
+import org.threadly.litesockets.SocketExecuterInterface.WireProtocol;
 import org.threadly.litesockets.utils.MergedByteBuffers;
 import org.threadly.litesockets.utils.SimpleByteStats;
 
+
 /**
- * This is the main Client object used in litesockets.  Anything that reads or writes data
- * will use this object.  The clients work by having buffered reads and writes for the socket.
+ * <p>This is the main Client interface used in litesockets.  Anything that reads or writes data
+ * will use this object.  The clients work by having buffered reads and writes for the socket. They tell the
+ * SocketExecuter when a read can be added or a write is ready for the socket.</p>
  * 
- * All reads will issue on a callback on a Reader in a single threaded manor.
+ * <p>All reads are issued on a callback on a Reader in a single threaded manor.
  * All writes will be sent in the order they are received.  In general its better to write full protocol parsable
  * packets with each write where possible.  If not possible your own locking/ordering will have to be ensured.
  * Close events will happen on there own Closer callback but it uses the same ThreadKey as the Reader thread.  
- * All Reads should be received before a close event happens.
+ * All Reads should be received before a close event happens.</p>
  * 
- * The client object can not functions with out being in a SocketExecuter.
+ * The client object can not function with out being in a SocketExecuter.  Writes can be added before its put
+ * in the executer, but it will not write to the socket until added to the executer.
  * 
  * @author lwahlmeier
  *
  */
-public abstract class Client {
-  public static final int DEFAULT_MAX_BUFFER_SIZE = 64*1024;
-  public static final int MIN_READ= 4*1024;
-  
-  private final AtomicInteger readBufferSize = new AtomicInteger(0);
-  private final AtomicInteger writeBufferSize = new AtomicInteger(0);
-  
-  protected ClientByteStats stats = new ClientByteStats();
-  
-  protected int maxBufferSize = DEFAULT_MAX_BUFFER_SIZE;
-  protected int minAllowedReadBuffer = MIN_READ;
-  
-  private final MergedByteBuffers readBuffers = new MergedByteBuffers();
-  private final MergedByteBuffers writeBuffers = new MergedByteBuffers();
-  private ByteBuffer currentWriteBuffer;
-  private ByteBuffer readByteBuffer = ByteBuffer.allocate(maxBufferSize*2);
-  
-  protected volatile Closer closer;
-  protected volatile Reader reader;
-  protected volatile SubmitterExecutorInterface sei;
-  protected volatile SocketExecuterBase ce;
-  protected AtomicBoolean closed = new AtomicBoolean(false);
+public interface Client {
   
   /**
-   * Returns the SimpleByteStats object for this client.
+   * Returns true if this client can have reads added to it or false if its read buffers are full.
```

Comment:
If you want the javadocs formatted a little better change true -> {@code true} (this is one I had to update all over in threadly's javadocs